pytorch_transformers/tokenization_roberta.py

```diff
@@ -0,0 +1,220 @@
+        to this instance.
+        """
+        if isinstance(f, six.string_types):
+            try:
+                if not ignore_utf_errors:
+                    with open(f, 'r', encoding='utf-8') as fd:
+                        self.add_from_file(fd)
+                else:
+                    with open(f, 'r', encoding='utf-8', errors='ignore') as fd:
+                        self.add_from_file(fd)
+            except FileNotFoundError as fnfe:
+                raise fnfe
+            except UnicodeError:
+                raise Exception("Incorrect encoding detected in {}, please "
+                                "rebuild the dataset".format(f))
+            return
+
+        lines = f.readlines()
+        for line in lines:
+            idx = line.rfind(' ')
+            if idx == -1:
+                raise ValueError("Incorrect dictionary format, expected '<token> <cnt>'")
+            word = line[:idx]
+            count = int(line[idx + 1:])
+            self.indices[word] = len(self.symbols)
+            self.symbols.append(word)
+            self.count.append(count)
+    
+    def encode_line(self, line, line_tokenizer=tokenize_line, add_if_not_exist=True,
+                    consumer=None, append_eos=True, reverse_order=False):
+        words = line_tokenizer(line)
+        if reverse_order:
+            words = list(reversed(words))
+        nwords = len(words)
+        ids = [0] * (nwords + 1 if append_eos else nwords)
+
+        for i, word in enumerate(words):
+            if add_if_not_exist:
+                idx = self.add_symbol(word)
+            else:
+                idx = self.index(word)
+            if consumer is not None:
+                consumer(word, idx)
+            ids[i] = idx
+        if append_eos:
+            ids[nwords] = self.eos_index
+        return ids
+
+
+
+
+class RobertaTokenizer(PreTrainedTokenizer):
+    """
+    RoBERTa tokenizer. Peculiarities:
+        - GPT-2 tokenizer with a different integer mapping on top.
+    """
+    vocab_files_names = VOCAB_FILES_NAMES
+    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+
+    def __init__(self, dict_file,
+                 bos_token="<s>", eos_token="</s>", **kwargs):
+        super(RobertaTokenizer, self).__init__(bos_token=bos_token, eos_token=eos_token, **kwargs)
+
+        self.gpt2_tokenizer = GPT2Tokenizer.from_pretrained('gpt2')
+        self.dictionary = Dictionary.load(dict_file)
+
+    def _tokenize(self, text):
+        """ Use GPT-2 Tokenizer """
+        return self.gpt2_tokenizer._tokenize(text)
+
+    def encode(self, text):
+        """ Converts a string in a sequence of ids (integer), using the tokenizer and vocabulary.
+        """
+        gpt2_tokens_joined = " ".join(
+            str(x) for x in self.gpt2_tokenizer.convert_tokens_to_ids(self.tokenize(text))
+        )
+        bpe_sentence = '<s> ' + gpt2_tokens_joined + ' </s>'
```

Comment:
I'm not 100% sure, but should we remove the `<s>` here? It's more of a `[cls]` token than a beginning of sentence token. That way it'll also work for sentence-pair tasks:
```
[tokenizer.cls_token] + tokenizer.encode(text_a) + [tokenizer.sep_token] + tokenizer.encode(text_b)
```
will produce the desired format:
```
<s> a b c </s> </s> 1 2 3 </s>
```